doc-deploy-dev-index/action.yml

```
@@ -0,0 +1,118 @@
+
+  cname:
+    description: >
+      The Canonical Name (CNAME) that points to the documentation website.
+    required: true
+    type: string
+
+  index-name:
+    description: >
+      The identifier given to the document in pymeilisearch.
+    required: true
+    type: string
+
+  api-key:
+    description: >
+      The API key used to access the Meilsearch instance host.
+    required: true
+    type: string
+
+  host-url:
+    description: >
+      The URL where the Meilsearch instance is hosted.
+    required: true
+    type: string
+
+  python-version:
+    description: >
+      Python version used for execution of the stable docs scraping.
+    required: true
+    type: string
+
+  # Optional inputs
+
+  doc-artifact-name:
+    description: >
+      The name of the HTML documentation artifact. This artifact is expected to
+      contain all the HTML and static files.The dafault value is ``documentation-html``.
+    required: false
+    default: 'documentation-html'
+    type: string
+
+  template:
+    description: >
+      The "template" parameter specifies the layout used for the HTML documentation.
+      By default, it is set to ``sphinx_pydata`` which assumes that the document is
+      constructed using the pydata-sphinx-theme or its associated theme,
+      such as the ansys-sphinx-theme.
+    required: false
+    default: "sphinx_pydata"
+    type: string
+
+
+
+runs:
+  using: "composite"
+  steps:
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Download the documentation artifact from the current workflow.
+
+    - name: "Download the development documentation artifact"
+      uses: actions/download-artifact@v3
+      with:
+        name: ${{ inputs.doc-artifact-name }}
+        path: ${{ inputs.doc-artifact-name }}
+
+    - name: "Display structure of ${{ inputs.doc-artifact-name }}"
+      shell: bash
+      run: |
+        ls -R ${{ inputs.doc-artifact-name }}
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Install the pymeilisearch required for the indexing.
+
+    - name: "Set up Python"
+      uses: ansys/actions/_setup-python@main
+      with:
+        python-version: '3.10'
+
+    - name: "Install build and twine"
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip pymeilisearch
+
+     # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Scrap the document and deploy it to pymeilisearch.
+
+    - name: Scrape the dev documentation to meilisearch
+      shell: bash
+      run: |
+        pymeilisearch upload --template ${{ inputs.template }} --index ${{ inputs.library_name }}-vdev  --cname ${{ inputs.cname }}/version/dev/ html ${{ inputs.doc-artifact-name }}
```

The index is being passed as argument:

```suggestion
        pymeilisearch upload --template ${{ inputs.template }} --index ${{ inputs.index-name }}  --cname ${{ inputs.cname }}/version/dev/ html ${{ inputs.doc-artifact-name }}
```